s3/pkg/service/object.go

```diff
@@ -316,11 +318,38 @@ func (s *s3Service) GetObjectMeta(ctx context.Context, in *pb.Object, out *pb.Ge
 		return err
 	}
 
+	// get expiration date
+	if in.IsHeadReq == true {
+		out.ExpireTime, out.RuleId = getObjectExpDate(bucket.LifecycleConfiguration, object.ObjectKey, object.LastModified)
+	}
+
 	out.Object = object.Object
 	object.StorageClass, _ = GetNameFromTier(object.Tier, utils.OSTYPE_OPENSDS)
 	return nil
 }
 
+func getObjectExpDate(lcRules []*pb.LifecycleRule, objKey string, lastModified int64) (int64, string) {
+	var expTime int64 = 0
+	for _, r := range lcRules {
+		if r.Status == utils2.RuleStatusDisabled {
+			continue
+		}
+		for _, ac := range r.Actions {
+			if ac.Name == utils2.ActionNameExpiration {
+				if r.Filter.Prefix == "" || strings.HasPrefix(objKey, r.Filter.Prefix) {
+					exp := lastModified + 60*60*24*int64(ac.Days)
+					if expTime < exp {
```

Comment:
When can it land in this situation? i.e. expTime < exp

Reply:
If there are more than one expiration rule, but expiration days are not the same, we will remove object when the latest expiration time arrived.

Reply:
Got it